internal/clock/clock.go

```diff
@@ -44,12 +48,19 @@ func NewMock() *Mock {
 
 // Add moves the current time of the mock clock forward by the duration.
 // This should only be called from a single goroutine at a time.
-func (m *Mock) Add(d time.Duration) {
+func (m *Mock) Add(d time.Duration, condition *gopeek.Condition) {
 	m.Lock()
 	// Calculate the final time.
 	end := m.now.Add(d)
 
 	for len(m.timers) > 0 && m.now.Before(end) {
+		// Wait before popping from timers
```

Comment:
Can you explain why we need to wait before popping timers? I would expect a wait after popping a timer.